stdlib/private/OSLog/OSLogMessage.swift

```diff
@@ -0,0 +1,309 @@
+  ///    The default is public.
+  public mutating func appendInterpolation(
+    _ number: @autoclosure @escaping () -> Int,
+    format: IntFormat = .decimal,
+    privacy: Privacy = .public
+  ) {
+    guard argumentCount < maxOSLogArgumentCount else { return }
+
+    addIntHeadersAndFormatSpecifier(
+      format,
+      isPrivate: privacy == .private,
+      bitWidth: Int.bitWidth,
+      isSigned: true)
+    arguments.append(number)
+  }
+
+  /// Construct/update format string and headers from the qualifiers (of the
+  /// interpolated expression) passed as parameters.
+  ///
+  /// All arguments to this function must be known at compile time.
+  public mutating func addIntHeadersAndFormatSpecifier(
+    _ format: IntFormat,
+    isPrivate: Bool,
+    bitWidth: Int,
+    isSigned: Bool
+  ) {
+    formatString += getIntegerFormatSpecifier(
+      format,
+      isPrivate: isPrivate,
+      bitWidth: bitWidth,
+      isSigned: isSigned)
+
+    addArgumentHeaders(
+      // Always explicitly set either public or private flag bit.
+      flagAndType: isPrivate ? 1 : 2,
+      byteSize: UInt8(bitWidth / 8))
+
+    updateSummaryBytes(isPrivate: isPrivate)
+  }
+
+  // Update the summary bytes: preamble and argument count.
+  public mutating func updateSummaryBytes(isPrivate: Bool) {
+    preamble |= isPrivate ? 1 : 0
+    argumentCount += 1
+  }
+
+  // Append the given argument header and size.
+  public mutating func addArgumentHeaders(flagAndType: UInt8, byteSize: UInt8) {
+    arguments.append(flagAndType)
+    arguments.append(byteSize)
+  }
+
+  /// Construct an OS log format specifier from the given parameters.
+  /// All arguments to this function must be known at compile time.
+  public func getIntegerFormatSpecifier(
+    _ format: IntFormat,
+    isPrivate: Bool,
+    bitWidth: Int,
+    isSigned: Bool
+  ) -> String {
+    var formatSpecifier: String = isPrivate ? "%{private}" : "%{public}"
+
+    // Determine the length modifier of the format specifier.
+    switch bitWidth {
+    case _ where bitWidth == CInt.bitWidth:
+      break
+    case _ where bitWidth == CLongLong.bitWidth:
+      formatSpecifier += "ll"
+    default:
+      fatalError("Int size not supported")
+    }
+
+    switch (format) {
+    case .hex:
+      formatSpecifier += "x"
+    case .octal:
+      formatSpecifier += "o"
+    default:
+      formatSpecifier += isSigned ? "d" : "u"
+    }
+    return formatSpecifier
+  }
+}
+
+extension String {
+  /// Replace all percents "%" in the string by "%%" so that the string can be
+  /// interpreted as a C format string.
+  public var percentEscapedString: String {
+    get {
+      return self
+        .split(separator: "%", omittingEmptySubsequences: false)
+        .joined(separator: "%%")
+    }
+  }
+}
+
+public struct OSLogMessage :
+  ExpressibleByStringInterpolation, ExpressibleByStringLiteral
+{
+  public let oslogInterpolation : OSLogInterpolation
```

Comment:
No space after "oslogInterpolation". Also Think you can safely shorten this variable name to "interpolation".